R/setkey.R

```diff
@@ -86,12 +86,26 @@ setkeyv <- function(x, cols, verbose=getOption("datatable.verbose"), physical=TR
     if (!typeof(.xi) %chin% c("integer","logical","character","double")) stop("Column '",i,"' is type '",typeof(.xi),"' which is not supported as a key column type, currently.")
   }
   if (!is.character(cols) || length(cols)<1L) stop("Internal error. 'cols' should be character at this point in setkey; please report.") # nocov
-  if (verbose) {
-    tt = suppressMessages(system.time(o <- forderv(x, cols, sort=TRUE, retGrp=FALSE)))  # system.time does a gc, so we don't want this always on, until refcnt is on by default in R
-    # suppress needed for tests 644 and 645 in verbose mode
-    cat("forder took", tt["user.self"]+tt["sys.self"], "sec\n")
+  
+  # get existing index name if any
+  found_index <- NULL
+  if(!is.null(indices(x))){
+      found_index <- names(attributes(attributes(x)$index))
+      found_index <- gsub("^__","", found_index)
+  }
+  
+  # forder only if index is not present
+  if(!identical(found_index, cols)){
```

Comment:
This works if there is one index,  as the test tests. But there can be multiple indexes (each index is a set of columns). It needs to find if any of the indexes match `cols` and then use that one.  The test needs expanding for cases of multiple indexes and `cols` both existing and not existing when there are multiple (so it tests it picks up the correct one).

Reply:
fixed it.